app/controllers/workflows_controller.rb

```diff
@@ -35,11 +35,14 @@ def show
   end
 
   def update
-    if params['current-status'].present?
-      find_step_for_process.update(status: params['current-status'])
-    else
-      find_step_for_process.update(status: 'completed')
-    end
+    parser = ProcessParser.new(process_from_request_body)
+    step = find_step_for_process
+
+    return render plain: process_mismatch_error(parser), status: :bad_request if parser.process != params[:process]
+
+    return render plain: status_mismatch_error(step), status: :conflict if params['current-status'] && step.status != params['current-status']
+
+    step.update(parser.to_h)
```

Comment:
The prior implementation allowed PUT requests containing no XML body, which had the effect of setting the status to `completed`. Can we verify that use case? It no longer exists after this PR; we now require an XML payload to set workflow step attributes.